cli/utils/version/version_checker.go

```diff
@@ -0,0 +1,197 @@
+		return availableNewestVersions{}, err
+	}
+	stable, err := browseVersions(usedVersion, versionInfo.Stable)
+	if err != nil {
+		return availableNewestVersions{}, err
+	}
+	prerelease, err := browseVersions(usedVersion, versionInfo.Prerelease)
+	if err != nil {
+		return availableNewestVersions{}, err
+	}
+	return availableNewestVersions{stable: stable, prerelease: prerelease}, nil
+}
+
+func browseVersions(usedVersion *version.Version, versions []string) (newestVersions, error) {
+
+	newestCompatible := usedVersion
+	newestIncompatible := usedVersion
+
+	usedVersionSeg := usedVersion.Segments()
+	if len(usedVersionSeg) != 3 {
+		return newestVersions{}, errors.New("Unexpected number of segements")
+	}
+
+	for _, vString := range versions {
+		v, err := version.NewSemver(vString)
+		if err != nil {
+			return newestVersions{}, err
+		}
+
+		vSeg := v.Segments()
+		if len(vSeg) != 3 {
+			return newestVersions{}, errors.New("Unexpected number of segements")
+		}
+
+		if usedVersionSeg[0] == vSeg[0] && usedVersionSeg[1] == vSeg[1] {
+			// Major and minor version match (ensures compatible APIs)
+			if newestCompatible.Compare(v) < 0 {
+				newestCompatible = v
+			}
+		} else if newestIncompatible.Compare(v) < 0 {
+			newestIncompatible = v
+		}
+	}
+
+	newerVersions := newestVersions{}
+	if newestCompatible.Compare(usedVersion) > 0 {
+		newerVersions.newestCompatible = newestCompatible
+	}
+	if newestIncompatible.Compare(usedVersion) > 0 {
+		newerVersions.newestIncompatible = newestIncompatible
+	}
+
+	return newerVersions, nil
+}
+
+const checkInterval = time.Hour * 24
+
+type VersionChecker struct {
+	versionFetcherClient *versionFetcherClient
+}
+
+func NewVersionChecker() *VersionChecker {
+	versionChecker := VersionChecker{}
+	versionChecker.versionFetcherClient = newVersionFetcherClient()
+	return &versionChecker
+}
+
+// getNewerCLIVersion checks for newer CLI versions if the automatic version check is enabled in the config
+func (v *VersionChecker) getNewerCLIVersion(cliConfig config.CLIConfig, usedVersionString string) (availableNewestVersions, error) {
+
+	cliVersionInfo, err := v.versionFetcherClient.getCLIVersionInfo(usedVersionString)
+	if err != nil {
+		return availableNewestVersions{}, fmt.Errorf("error when fetching CLI version infos: %v", err)
+	}
+
+	res, err := getAvailableVersions(usedVersionString, *cliVersionInfo)
+	if err != nil {
+		return availableNewestVersions{}, fmt.Errorf("error when analyzing the available versions: %v", err)
+	}
+	return res, nil
+}
+
+const newCompatibleVersionMsg = `keptn version %s is available! Please visit https://keptn.sh for more information.`
+const newIncompatibleVersionMsg = `keptn version %s is available! Please note that this version is incompatible with your Keptn cluster` +
+	`version and requires to update the cluster too. Please visit https://keptn.sh for more information.`
+const disableMsg = `To disable this notice, run: 'keptn set config AutomaticVersionCheck false'`
```

Comment:
Please make the calling binary `keptn` a variable.

E.g., someone downloads the keptn cli and renames it into `keptn06` and even calling it from the local directory, then the text should be 
``To disable this notice, run: './keptn06 set config AutomaticVersionCheck false'``

Reply:
Good idea! Done.